disco_aws_automation/disco_metanetwork.py

```diff
@@ -236,3 +251,28 @@ def add_sg_rule(self, protocol, ports, sg_source=None, cidr_source=None):
         sg_args["to_port"] = ports[1]
         logging.debug("Adding sg_rule: %s", sg_args)
         self.vpc.vpc.connection.authorize_security_group(**sg_args)
+
+    def ip_by_offset(self, offset):
+        """
+        Pass in +10 and get 10th ip of subnet range
+        Pass in -2 and get 2nd to last ip of subnet
+
+        Returns IpAddress object, usually you'll want
+        to cast this to str.
+        """
+
+        try:
+            offset = int(offset)
+        except ValueError:
+            raise IPRangeError(
+                "Cannot find IP in metanetwork {0} by offset {1}."
+                .format(self.name, offset)
+            )
+
+        subnets = sorted(self.subnet_ip_networks)
+        base_address = subnets[0].first if offset >= 0 else subnets[-1].last
```

Comment:
Yep. The offset is picked from the first (or last) address of metanetwork. Here we just do a range check to make sure selected address is within range. The code that makes sure instance is put in the right subnet already existed: https://github.com/amplifylitco/asiaq/blob/master/disco_aws_automation/disco_aws.py#L188